### PR TITLE
Function App (Flex Cosumption) ZipDeploy publish support

### PR DESCRIPTION
**Pull request by anvillan**
### Issue describing the changes in this PR

resolves (TBD)

### In this PR:
 - Add new `UseBlobContainerDeploy` (`bool`) property to the `ZipDeploy` target/task to identify when to use the OneDeploy URL 
 "`<publishUrl>/api/publish`" endpoint to publish a Flex Consumption function app
 - When polling for the ZipDeploy pubish status, append the `status_text` message (if defined) along with the status itself. E.g.: `"Deployment status is Failed: App setting WEBSITE_RUN_FROM_PACKAGE is not supported..."`.
 - Add UTs for `ZipDeployTask` and `ZipDeploymentStatus`

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information
This is a copy of https://github.com/Azure/azure-functions-dotnet-worker/pull/2688, created to be able to pass required checks failing due to lack of permission(s).

### CODE REVIEW

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.Publish.ZipDeploy.targets [hunk 1]
```diff
@@ -50,7 +50,8 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
       DeploymentPassword="$(Password)"
       SiteName="$(DeployIisAppPath)"
       PublishUrl="$(PublishUrl)"
-      UserAgentVersion="$(ZipDeployUserAgent)"/>
+      UserAgentVersion="$(ZipDeployUserAgent)"
+      UseBlobContainerDeploy="$(UseBlobContainerDeploy)"/>
```
Comment (kshyju) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.Publish.ZipDeploy.targets:54]: from where will the value of `UseBlobContainerDeploy` be set? Is that something VS will set before evaluating this msbuild target during publish?

Comment (anvillan) [reply sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.Publish.ZipDeploy.targets:54]: Yes. VS IDE creates a publish profile that uses this task, and that contains all the properties defined for it.

When VS creates a publish profile for Function App (Flex Consumption), `UseBlobContainerDeploy=True` will be included in the profile.

File: sdk/Sdk/Tasks/ZipDeploy/ZipDeployTask.cs [hunk 2]
```diff
@@ -89,13 +94,19 @@ internal async System.Threading.Tasks.Task<bool> ZipDeployAsync(string zipToPubl
...
+
+            // "<publishUrl>/api/zipdeploy?isAsync=true" or "<publishUrl>/api/publish?RemoteBuild=false"
+            zipDeployPublishUrl = $"{zipDeployPublishUrl}/{publishUriPath}";
+
             if (logMessages)
             {
                 Log.LogMessage(MessageImportance.High, String.Format(StringMessages.PublishingZipViaZipDeploy, zipToPublishPath, zipDeployPublishUrl));
             }
 
             // use the async version of the api
```
[the remark this answers is not in the document]
Comment (anvillan) [reply sdk/Sdk/Tasks/ZipDeploy/ZipDeployTask.cs]: Done

**Comment by anvillan** [comment]
@kshyju @fabiocav @nzthiago 

Please let me know if anything else is required so this PR is approved, and it merges.

**Review by kshyju — APPROVED** [review]
What are the plans to test this E2E with the VS profile you mentioned?

**Pull request merged**